### PR TITLE
Add X-Okta-Request-Id header to responses Resolves: OKTA-84899

### PR DESCRIPTION
**Pull request by ashishgupta-okta**
Readding because it was already reverted due to a premature commit to master.

this supercedes https://github.com/okta/okta.github.io/pull/206

Api docs for request debugging using the X-Okta-Request-Id.

Resolves: OKTA-84899

Primary Reviewer: @mystiberry-okta @hassenkaraa-okta 
Reviewer : @federations-okta

### CODE REVIEW

**Comment by cwu-okta** [comment]
@mystiberry-okta please merge this for the preview release going out 4/4
The original PR already received the ship.

File: _source/docs/api/resources/oidc.md [hunk 1]
```diff
@@ -421,4 +423,4 @@ This API endpoint returns the OpenID Connect related metadata that can be used b
     ]
 }
 
-~~~~
\ No newline at end of file
+~~~~
```
Comment (larsjohansen-okta) [inline _source/docs/api/resources/oidc.md]: just curious; what happened here?

File: _source/docs/api/getting_started/design_principles.md [hunk 2]
```diff
@@ -266,7 +266,19 @@ X-Rate-Limit-Reset: 1366037820
 
 If the rate limit is exceeded, an HTTP 429 Status Code is returned.  The current Rate Limit is on a per-org per-endpoint basis.
 
-**Rate limits are enforced for all organizations.**
+## Request Debugging
+
+The requestId will always be present in every API response and can be used for debugging. This value can be used to correlate events from the Events API as well as the System Log events.
```
Comment (tthompson-okta) [inline _source/docs/api/getting_started/design_principles.md]: Nit: requestId looks like a variable name here. I think we should spell it out in plain english, i.e. `request ID`

File: _source/docs/api/getting_started/design_principles.md [hunk 3]
```diff
@@ -266,7 +266,19 @@ X-Rate-Limit-Reset: 1366037820
...
+
+The following header is set in each response:
+
+`X-Okta-Request-Id` - The UUID for the API request
```
Comment (tthompson-okta) [inline _source/docs/api/getting_started/design_principles.md]: I don't think we should document the internal structure (UUID). We want the flexibility to change this format in the future. I'd just say `The unique identifier for the API request`.

File: _source/docs/api/resources/oidc.md [hunk 4]
```diff
@@ -257,6 +257,7 @@ HTTP/1.1 401 Unauthorized​
 Cache-Control: no-cache, no-store​
 Pragma: no-cache​
 Expires: 0​
+X-Okta-Request-Id: reqVy8wsvmBQN27h4soU3ZEnA
```
Comment (tthompson-okta) [inline _source/docs/api/resources/oidc.md]: Why add this to a couple of random examples in the OIDC doc? I think we'd want to be consistent across the API docs (i.e. everywhere or nowhere). I think it just adds noise to the examples, so I'm in favor of nowhere.

File: _source/docs/api/getting_started/design_principles.md (hunk 2 above)
Comment (tthompson-okta) [inline _source/docs/api/getting_started/design_principles.md]: Add a hyperlink to the [Events API] - http://developer.okta.com/docs/api/resources/events.html

Comment (tthompson-okta) [inline _source/docs/api/getting_started/design_principles.md]: Add a Jira to support searching/filtering the Events API by requestId. It looks like we dont' support this right now, but inevitably someone is going to ask for it soon now that we're exposing the request ID.

File: _source/docs/api/getting_started/design_principles.md [hunk 5]
```diff
@@ -266,7 +266,21 @@ X-Rate-Limit-Reset: 1366037820
...
+
+The request ID will always be present in every API response and can be used for debugging. This value can be used to correlate events from the [Events API](http://developer.okta.com/docs/api/resources/events.html) as well as the System Log events.
+
+The following header is set in each response:
+
+`X-Okta-Request-Id` - The unique identifier for the API request
+
+~~~ http
+HTTP/1.1 200 OK
+X-OKta-Request-ID: reqVy8wsvmBQN27h4soUE3ZEnA
```
Comment (karlmcguinness-okta) [inline _source/docs/api/getting_started/design_principles.md]: `X-Okta-Request-ID` not `X-OKta-Request-ID` right?
